### PR TITLE
Add Interval.map

### PR DESCRIPTION
**Pull request by texane**
Add a class method returning an expression that maps values from the first interval into the second interval. First pull request for this project, let me know if it needs reworking or redundant with existing.

### CODE REVIEW

**Comment by asmeurer** [comment]
Please add tests.

**Comment by asmeurer** [comment]
I would make it so that `x` is a required argument.

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -799,6 +799,37 @@ def Ropen(cls, a, b):
...
+
+        Examples
+        ========
+
+        >> from sympy import Interval
+        >> from sympy.abc import a, b, x
+        >> from sympy import factor
+        >> e = Interval.map(Interval(10, 12), Interval(-1, 1))
+        >> print(e.subs(x, 10))
+        0
```
Comment (asmeurer) [inline sympy/sets/sets.py]: Shouldn't 10 map to -1 and 12 map to 1?

Comment (texane) [inline sympy/sets/sets.py]: Hi,

actually, there was a typo. 10 maps to -1, 11 to 0 and 12 to 1.
I fixed it in my latest branch.

Also, I made x a required argument, and added
examples/beginner/map_intervals.py


2015-11-16 22:30 GMT+01:00 Aaron Meurer <notifications@github.com>:

**Comment by oscarbenjamin** [comment]
I don't think that this `map` method is needed on Interval. It's not clear what the use case is but it doesn't seem like a core need that should be part of the (current) `Interval` class.

Thank you for sending this PR but I'm going to close this now. Feel free to open a new PR if you still think that this is needed.

**Pull request closed without merging**